src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -544,6 +544,12 @@ public MemoryBlock getOriginalPage(long pagePlusOffsetAddress) {
       assert (page != null);
       assert (page.getBaseObject() != null);
       return page;
+    } else if (tungstenMemoryMode == MemoryMode.OFF_HEAP) {
```

Comment:
Only difference between on-heap and off-heap is one more check baseobj in L545?

Reply:
Indeed it's yes. Maybe we remove this assert (page.getBaseObject() != null) and unify the paths together?

Reply:
yes. we should

Reply:
done.